experimental/sql/src/impl_to_iterators.py

```diff
@@ -147,7 +147,35 @@ class YieldRewriter(RelImplRewriter):
 
   @op_type_rewrite_pattern
   def match_and_rewrite(self, op: RelImpl.Yield, rewriter: PatternRewriter):
-    rewriter.replace_matched_op(Return.get(*op.ops))
+    if isinstance(op.parent_op(), RelImpl.Select):
+      rewriter.replace_matched_op(Return.get(*op.ops))
+    else:
+      assert (isinstance(op.parent_op(), RelImpl.Project))
+      for i, o in zip(range(len(op.ops)), op.ops):
+        rewriter.insert_op_before_matched_op(
+            LLVMInsertValue.build(
+                operands=[op.parent_block().args[0], o],
+                attributes={
+                    "position":
+                        ArrayAttr.from_list(
+                            [IntegerAttr.from_index_int_value(i)])
+                },
+                result_types=[
+                    self.convert_bag(op.parent_op().results[0].typ).types
+                ]))
+      rewriter.replace_matched_op(Return.get(op.parent_block().args[0]))
```

Comment:
There is certainly a broader conversation to be had for cases where the input and output types of the map do not match, since in those cases, we might have to allocate memory.

Reply:
Okay, in that case it will be simpler, i.e., I just need to add the alloca operation to the llvm mirror :+1:

Reply:
I don't even think you need that... `llvm.mlir.undef` can create "empty" structs.